javaslang/src/test/java/javaslang/collection/JavaConverters_ListViewMutableTest.java

```diff
@@ -0,0 +1,246 @@
+        // .hasMessage("Index: -1, Size: 1");
+    }
+
+    @Test
+    public void shouldThrowWhenAddingElementAtNonExistingIndexToNonEmpty() {
+        assertThatThrownBy(() -> of(1).add(2, null))
+                .isInstanceOf(IndexOutOfBoundsException.class);
+        // .hasMessage("Index: 2, Size: 1");
+    }
+
+    @Test
+    public void shouldAddNullToEmptyAtIndex0() {
+        final java.util.List<Object> list = empty();
+        list.add(0, null);
+        assertThat(list).isEqualTo(javaList((Object) null));
+    }
+
+    @Test
+    public void shouldAddNonNullToEmptyAtIndex0() {
+        final java.util.List<Integer> list = empty();
+        list.add(0, 1);
+        assertThat(list).isEqualTo(javaList(1));
+    }
+
+    @Test
+    public void shouldThrowWhenAddToEmptyAtNegativeIndex() {
+        assertThatThrownBy(() -> empty().add(-1, 1))
+                .isInstanceOf(IndexOutOfBoundsException.class);
+        // .hasMessage("Index: -1, Size: 0");
+    }
+
+    @Test
+    public void shouldThrowWhenAddToEmptyAtIndexGreaterSize() {
+        assertThatThrownBy(() -> empty().add(1, 1))
+                .isInstanceOf(IndexOutOfBoundsException.class);
+        // .hasMessage("Index: 1, Size: ");
+    }
+
+    // -- addAll(Collection)
+
+    @Test
+    public void shouldReturnFalseIfAddAllEmptyToEmptyListView() {
+        assertThat(empty().addAll(javaList())).isFalse();
+    }
+
+    @Test
+    public void shouldReturnFalseIfAddAllEmptyToNonEmptyListView() {
+        final java.util.List<Integer> list = of(1);
+        assertThat(list.addAll(javaList())).isFalse();
+        assertThat(list).isEqualTo(of(1));
+    }
+
+    @Test
+    public void shouldReturnTrueIfAddAllNonEmptyToEmptyListView() {
+        final java.util.List<Integer> list = empty();
+        final java.util.List<Integer> javaList = javaList(1);
+        assertThat(list.addAll(javaList)).isTrue();
+        assertThat(list).isEqualTo(javaList);
+    }
+
+    @Test
+    public void shouldReturnTrueIfAddAllNonEmptyToNonEmptyListView() {
+        final java.util.List<Integer> list = of(1, 2, 3);
+        assertThat(list.addAll(javaList(1, 2, 3))).isTrue();
+        assertThat(list).isEqualTo(javaList(1, 2, 3, 1, 2, 3));
+    }
+
+    // TODO
+
+    // -- TODO: addAll(int, Collection)
+
+    // -- TODO: clear()
+
+    // -- contains(Object)
+
+    @Test
+    public void shouldRecognizeListContainsSelf() {
+        final java.util.List<Object> list = empty();
+        list.add(list);
+        assertThat(list.contains(list)).isTrue();
+    }
+
+    // -- containsAll(Collection)
+
+    @Test
+    public void shouldRecognizeListContainsAllSelf() {
+        final java.util.List<Object> list = empty();
+        list.add(list);
+        assertThat(list.containsAll(list)).isTrue();
+    }
+
+    // -- equals(Object)
+
+    @Test
+    public void shouldRecognizeSelfEqualityOfListThatContainsItself() {
+        final java.util.List<Object> list = empty();
+        list.add(list);
```

Comment:
\[FindBugs\] INFO: IL: A collection is added to itself in javaslang\.collection\.JavaConverters\_ListViewMutableTest\.shouldRecognizeSelfEqualityOfListThatContainsItself\(\)